### PR TITLE
Refactor the BDII provider

### PR DESCRIPTION
**Pull request by brianhlin**
Will make it easier to address issues in #364. I tested this against our local cluster and things look with a quick diff (number of lines is the same but there are some differences that look simply like slight changes in LDIF entry ordering).

@aleita @kreczko @andiri @lfield would you guys be able to verify output of the refactored provider in this PR? For now, I'm going to move forward on providing multi collector support assuming that this PR is acceptable!

### CODE REVIEW

**Comment by lfield** [comment]
I get the following exception:
`Traceback (most recent call last):
  File "./provider-new", line 421, in <module>
    main()
  File "./provider-new", line 415, in main
    running_jobs))
  File "./provider-new", line 233, in format_share_entry
    resource_keys=resource_keys)
KeyError: 'share_fqdn'
`

**Comment by aleita** [comment]
+1

**Comment by brianhlin** [comment]
Argh, sorry about that! Turns out my CE config wasn't correct so there were a few issues that I missed but they should all be fixed now!

**Comment by lfield** [comment]
Failed communication with CE collector on ce8test.cern.ch:9619
Traceback (most recent call last):
  File "./htcondor-ce-provider", line 421, in <module>
    main()
  File "./htcondor-ce-provider", line 392, in main
    ce_schedd_ad = get_ce_schedd_ad(ce_fqdn)
  File "./htcondor-ce-provider", line 252, in get_ce_schedd_ad
    return ce_schedd_ad
UnboundLocalError: local variable 'ce_schedd_ad' referenced before assignment

**Comment by aleita** [comment]
last commit it's fine for me

[root@ce-05 ~]# echo $?
0

**Comment by brianhlin** [comment]
@lfield should be fixed

**Comment by lfield** [comment]
It now runs but I have to study the differences in output.

**Review by lfield — APPROVED** [review]
Looks good to me. I compared the content with our production system and it seems fine.

**Comment by lfield** [comment]
Looks good to me. Here are two tips for comparing content using ldapsearch.

```
-o ldif-wrap=no # Doesn't wrap lines
-S GLUE2UniqueID # Orders entries by attribute
```

File: contrib/bdii/htcondor-ce-provider [hunk 1]
```diff
@@ -150,6 +153,165 @@ def handler(signum, frame):
...
+                                issuer=issuer,
+                                central_manager=central_manager)
+
+
+def format_share_entry(central_manager, ce_fqdn, voname, topologies, total_jobs, idle_jobs, running_jobs):
+    resource_keys = ""
+    for machine_ad in topologies:
+        resource = format_resource_name(machine_ad)
+        resource_keys += RESOURCE_KEY.format(central_manager=central_manager, resource=resource)
+        return SHARE_LDIF.format(ce_fqdn=ce_fqdn,
```
[the remark this answers is not in the document]
Comment (brianhlin) [reply contrib/bdii/htcondor-ce-provider]: Yup, this got fixed in https://github.com/htcondor/htcondor-ce/pull/387/commits/7778a43b8ecbfddec2277665a168327b0ea5c872

**Comment by lfield** [comment]
Will this be released or do you plan to do more changes?

**Comment by brianhlin** [comment]
Hi @lfield I'd like to fit in a fix for #364 before a release as well as Python 3 support. It's not looking like this will be done before HTCondor Week Europe next week so I'm shooting for early October.

**Comment by lfield** [comment]
ok, I will leave this in qa and push it to prod once the other changes have been done. Ping me when you have something to test.

**Pull request merged**